### PR TITLE
Add `position_ids` in `XLMRobertaXLForCausalLM.prepare_inputs_for_generation`

### PR DESCRIPTION
**Pull request by ydshieh**
# What does this PR do?

The model component `XLMRobertaXLEmbeddings` will perform

> position_ids = create_position_ids_from_input_ids(input_ids, self.padding_idx, past_key_values_length)

if `position_ids` is not passed to the model (say `XLMRobertaXLForCausalLM`).

However, during the decoding, even if `past_key_values_length` is passed, the `create_position_ids_from_input_ids` will compute the wrong `position_ids` when the `input_ids` is only the single last token. This is because `create_position_ids_from_input_ids` relies the whole  `input_ids` in order to get the number of padding tokens:

>     mask = input_ids.ne(padding_idx).int()

(since `past_key_values_length` doesn't contain this information.)

In general, we need to prepare `position_ids` in `prepare_inputs_for_generation`, like what has been done in

> GenerationMixin.prepare_inputs_for_generation

This PR adds this for `XLMRobertaXLForCausalLM` (whose `prepare_inputs_for_generation` is already overwritten from the parent class)

---------------------------------------------------------------------------------------------------------------------------------

As a bonus:

> tests/models/xlm_roberta_xl/test_modeling_xlm_roberta_xl.py::XLMRobertaXLModelTest::test_assisted_decoding_matches_greedy_search_1_same

is also no longer flaky (running 10000 times) while it was failing 5% of the time before this PR.

### CODE REVIEW

File: src/transformers/models/xlm_roberta_xl/modeling_xlm_roberta_xl.py [hunk 1]
```diff
@@ -1119,6 +1119,12 @@ def prepare_inputs_for_generation(self, input_ids, past_key_values=None, attenti
         if attention_mask is None:
             attention_mask = input_ids.new_ones(input_shape)
 
+        # Create missing `position_ids` on the fly
+        position_ids = None
+        if model_kwargs.get("position_ids") is None:
+            position_ids = create_position_ids_from_input_ids(input_ids, padding_idx=self.config.pad_token_id)
+            position_ids = position_ids  # placed in kwargs for further processing (see below)
```
Comment (zucchini-nlp) [inline src/transformers/models/xlm_roberta_xl/modeling_xlm_roberta_xl.py:1126]: ultra nit: this line is redundant a bit

**Review by zucchini-nlp — APPROVED** [review]
Oh yeah, we have some inconsistencies with how position ids are created in models. I've been trying to create position ids in all models in the same way, but it got too big and complicated. We'll need to come back to position ids later at some point

 For now LGTM, thanks!

File: src/transformers/models/xlm_roberta_xl/modeling_xlm_roberta_xl.py (hunk 1 above)
Comment (ydshieh) [reply src/transformers/models/xlm_roberta_xl/modeling_xlm_roberta_xl.py:1126]: wow, i don't know why I added this 😭 thank you!

Comment (ydshieh) [reply src/transformers/models/xlm_roberta_xl/modeling_xlm_roberta_xl.py:1126]: BTW, some models' position ids are different (`Roberta` one is an example). It should use `create_position_ids_from_input_ids`, but `Roberta`'s `prepare_inputs_for_generation` doesn't use it and use the common one. 

This is another story - but I kind think it's a bug ...

Comment (zucchini-nlp) [reply src/transformers/models/xlm_roberta_xl/modeling_xlm_roberta_xl.py:1126]: yeah, that is the reason we need one method to `create_position_ids` which will be used everywhere. But last time it resulted in so many failures, especially when checking TF/JAX equivalence. Also might be quite breaking change :(

Oke, we'll take it noted for now

**Review by ArthurZucker — APPROVED** [review]
Looks good to me, let's cleanup a bit before merging 😉
